cmd/knode-manager/main.go

```diff
@@ -0,0 +1,26 @@
+package main
+
+import (
+	"context"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"k8s.io/component-base/cli"
+
+	"github.com/kosmos.io/kosmos/cmd/knode-manager/app"
+)
+
+func main() {
+	ctx, cancel := context.WithCancel(context.Background())
+	sig := make(chan os.Signal, 1)
+	signal.Notify(sig, syscall.SIGINT, syscall.SIGTERM)
+	go func() {
+		<-sig
+		cancel()
+	}()
```

Comment:
what does `sig` for

Reply:
> what does `sig` for

For keeping the main thread from exiting.

Reply:
ctx := apiserver.SetupSignalContext()
what about using this ？

Reply:
it's a better way